<launch>

  <arg name="nodelet_manager" default="false" />  <!-- can be set to an existing nodelet manager to be used for this nodelet -->
  <arg name="nodelet_name" default="SampleNodelet" /> <!-- name of the nodelet -->
  <arg name="custom_params" default="false" />  <!-- can be set to a custom parameter yaml file that shall be used to overwrite parameters of this package -->
  <arg name="create_publisher" default="true" />
  <arg name="create_subscriber" default="true" />

  <group ns="/sample_package_cpp">
    <!-- standalone nodelet -->
    <node if="$(eval nodelet_manager == false)" pkg="nodelet" type="nodelet" name="$(arg nodelet_name)" args="standalone sample_package_cpp/SampleNodelet" output="screen">
      <rosparam command="load" file="$(find sample_package_cpp)/launch/params_SampleNodelet.yaml" />  <!-- common parameters from config file -->
      <rosparam if="$(eval custom_params != false)" command="load" file="$(arg custom_params)" />
      <param name="create_publisher" value="$(arg create_publisher)"/>
      <param name="create_subscriber" value="$(arg create_subscriber)"/>
      <remap to="/message" from="~/message"/>  <!-- remaps a global topic to the topic of the node -->
    </node>

    <!-- nodelet with external nodelet manager -->
    <node if="$(eval nodelet_manager != false)" pkg="nodelet" type="nodelet" name="$(arg nodelet_name)" args="load sample_package_cpp/SampleNodelet $(arg nodelet_manager)" output="screen">
      <rosparam command="load" file="$(find sample_package_cpp)/launch/params_SampleNodelet.yaml" />
      <rosparam if="$(eval custom_params != false)" command="load" file="$(arg custom_params)" />
      <param name="create_publisher" value="$(arg create_publisher)"/>
      <param name="create_subscriber" value="$(arg create_subscriber)"/>
      <remap to="/message" from="~/message"/>  <!-- remaps a global topic to the topic of the node -->
    </node>
  </group>

</launch>

